packet/src/main/java/org/jclouds/packet/filters/AddApiVersionToRequest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.packet.filters;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jclouds.http.HttpException;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.http.HttpRequestFilter;
+import org.jclouds.rest.annotations.ApiVersion;
+
+import com.google.common.net.HttpHeaders;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Singleton
+public class AddApiVersionToRequest implements HttpRequestFilter {
+
+    private final String apiVersion;
+
+    @Inject
+    public AddApiVersionToRequest(@ApiVersion String apiVersion) {
+        this.apiVersion = checkNotNull(apiVersion, "apiVersion");
+    }
+
+    @Override
+    public HttpRequest filter(final HttpRequest request) throws HttpException {
+        return request.toBuilder()
+                .replaceHeader(HttpHeaders.ACCEPT, String.format("application/json; version=%s", apiVersion))
```

Comment:
Do we want to override the Accept header, whatever it is? Or just append the version if missing? Both approaches LGTM.
If we go for the former, then we could remove the `@Consumes` annotations from the Api classes, since the header will be always put by this filter.